src/clang-c-frontend/expr2ccode.cpp

```diff
@@ -0,0 +1,1509 @@
+      greaterthan2tc gt2(
+        add2tc(same->side_1->type, same->side_1, gen_ulong(1)),
+        add2tc(arr_type->subtype, addr->ptr_obj, arr_size));
+      or2tc in_bounds(gt, gt2);
+      simplify(in_bounds);
+      return convert(migrate_expr_back(in_bounds), precedence);
+    }
+  }
+
+  equality2tc eq(same->side_1, same->side_2);
+  return convert(migrate_expr_back(not2tc(eq)), precedence);
+}
+
+std::string expr2ccodet::convert_pointer_offset(
+  const exprt &src,
+  unsigned &precedence [[maybe_unused]])
+{
+  std::string dest = "pointer_offset";
+  dest += '(';
+
+  forall_operands(it, src)
+  {
+    unsigned p;
+    std::string op = convert(*it, p);
+
+    if(it != src.operands().begin())
+      dest += ", ";
+
+    dest += op;
+  }
+
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_malloc(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0;
+  std::string op0 = convert(src.op0(), p0);
+
+  std::string dest = "malloc";
+  dest += '(';
+  dest += op0;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_realloc(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0, p1;
+  std::string op0 = convert(src.op0(), p0);
+  std::string size = convert((const exprt &)src.cmt_size(), p1);
+
+  std::string dest = "realloc";
+  dest += '(';
+  dest += op0;
+  dest += ", ";
+  dest += size;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_alloca(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 1)
+    return convert_norep(src, precedence);
+
+  unsigned p0;
+  std::string op0 = convert(src.op0(), p0);
+
+  std::string dest = "alloca";
+  dest += '(';
+  dest += op0;
+  dest += ')';
+
+  return dest;
+}
+
+std::string expr2ccodet::convert_nondet(const exprt &src, unsigned &precedence)
+{
+  if(src.operands().size() != 0)
+    return convert_norep(src, precedence);
+
+  std::string type_str = "";
+
+  if(src.type().is_bool())
+  {
+    type_str += "bool";
+  }
+  else if(src.type().id() == "signedbv" || src.type().id() == "unsignedbv")
+  {
+    BigInt width = string2integer(src.type().width().as_string());
+
+    if(src.type().id() == "unsignedbv")
+      type_str += "u";
+
+    if(width == config.ansi_c.int_width)
+      type_str += "int";
+
+    if(width == config.ansi_c.long_int_width)
+      type_str += "long";
+
+    if(width == config.ansi_c.char_width)
+      type_str += "char";
+
+    if(width == config.ansi_c.short_int_width)
+      type_str += "short";
+  }
+  else if(src.type().id() == "floatbv" || src.type().id() == "fixedbv")
+  {
+    BigInt width = string2integer(src.type().width().as_string());
+
+    if(width == config.ansi_c.single_width)
+      type_str += "float";
+    if(width == config.ansi_c.double_width)
+      type_str += "double";
```

Comment:
Same as above: what about `long double` and other bit-widths? Maybe add an assertion and/or a comment.

Reply:
Same as above. There does not seem to be a case of a `nondet` for `long double` in `esbmc_intrinsics.h`. I've added a comment in the description on the method.